scripts/o3de/o3de/manifest.py

```diff
@@ -156,25 +156,28 @@ def get_default_o3de_manifest_json_data() -> dict:
     return json_data
 
 def get_o3de_manifest() -> pathlib.Path:
-    manifest_path = get_o3de_folder() / 'o3de_manifest.json'
-    if not manifest_path.is_file():
-        json_data = get_default_o3de_manifest_json_data()
-
-        with manifest_path.open('w') as s:
-            s.write(json.dumps(json_data, indent=4) + '\n')
-
-    return manifest_path
+    return get_o3de_folder() / 'o3de_manifest.json'
 
 
 def load_o3de_manifest(manifest_path: pathlib.Path = None) -> dict:
     """
     Loads supplied manifest file or ~/.o3de/o3de_manifest.json if None
+    If the supplied manifest_path is None and  ~/.o3de/o3de_manifest.json doesn't exist default manifest data
+    is instead returned.
+    Note: There is a difference between supplying a manifest_path parameter of None and '~/.o3de/o3de_manifest.json'
+    In the former if the o3de_manifest.json doesn't exist, default o3de manifest data is returned.
+    In the later that the o3de_manifest.json must exist as the caller explicitly specified the manifest path
 
     raises Json.JSONDecodeError if manifest data could not be decoded to JSON
     :param manifest_path: optional path to manifest file to load
     """
     if not manifest_path:
         manifest_path = get_o3de_manifest()
+        # If the default o3de manifest file doesn't exist and the manifest_path parameter was not supplied
+        # return the default o3de manifest data
+        if not manifest_path.is_file():
+            return get_default_o3de_manifest_json_data()
```

Comment:
Throughout, each time a condition for the default is triggered feels like an exceptional case. It may be appropriate to log when the default gets used.

Reply:
I'll add an information log about it.
It is not an error, but it should occur only when the manifest file doesn't exist.